src/aws/osml/image_processing/gdal_tile_factory.py

```diff
@@ -339,31 +339,85 @@ def _read_from_rlevel_as_array(
+        else:
+            return pixel_array
+
+    def _normalize_bands(
+        self, pixel_array: np.array, normalize_band_func: Callable[[gdal.Band, np.array], np.array]
+    ) -> np.array:
+        if pixel_array.ndim == 2:  # 1-band grayscale image
+            band = self.raster_dataset.GetRasterBand(1)
+            normalized_pixels = normalize_band_func(band, pixel_array)
+        elif pixel_array.ndim == 3 and pixel_array.shape[2] >= 3:
+            # Multiband image, select the first 3 bands
             pixel_array = pixel_array[:, :, 0:3]
+            band_count = 3
+            normalized_bands = []
+            for band_idx in range(band_count):
+                band = self.raster_dataset.GetRasterBand(band_idx + 1)
+                normalized_bands.append(normalize_band_func(band, pixel_array[:, :, band_idx]))
+
+            # Combine the normalized bands into a single image
+            normalized_pixels = np.stack(normalized_bands, axis=2)
+        else:
+            logger.debug("Skipping normalization.")
+            normalized_pixels = pixel_array
+        return normalized_pixels
+
+    @staticmethod
+    def _normalize_band_minmax(band: gdal.Band, pixel_array: np.array) -> np.array:
+        """
+        This method applies Min-Max normalization to an individual band. It attempts to us the min-max values from the
+        image but if they are none it reverts to using values from the individual tile.
+
+        :param pixel_array: the input image pixels
+        :return: a Min-Max normalized 8-bit per pixel image
+        """
+        # Get the minimum and maximum values from the entire GDAL dataset
+        min_value = band.GetMinimum()
+        max_value = band.GetMaximum()
+
+        # If the minimum and maximum values are not available, calculate them from the tile
+        if min_value is None:
+            min_value = np.min(pixel_array)
+        if max_value is None:
+            max_value = np.max(pixel_array)
+
+        # Apply the Min-Max normalization
+        normalized_pixel = (pixel_array - min_value) * (255.0 / max(max_value, 1.0))
+        normalized_pixel = np.clip(normalized_pixel, 0.0, 255.0)
+        return normalized_pixel.astype(np.uint8)
+
+    @staticmethod
+    def _normalize_band_dra(band: gdal.Band, pixel_array: np.array) -> np.array:
+        """
+        This method performs DRA on an input pixel_array using gdal Band data. Normalization is performed with
+        respect to the entire image using the GDAL histogram.
 
-        max_channel_value = np.max(pixel_array)
-        hist, bin_edges = np.histogram(pixel_array.ravel(), bins=max_channel_value + 1, range=(0, max_channel_value))
-        dra_parameters = DRAParameters.from_counts(hist.tolist(), max_percentage=0.97)
-        for_display = (
+        :param pixel_array: the input image pixels
+        :return: a range adjusted 8-bit per pixel image
+        """
+        hist = band.GetHistogram(min=0, max=255, buckets=256)
```

Comment:
The max value may be greater than 255. This needs to be the maximum value of a pixel value in the original image. The original image may be an 11-16 bit per pixel panchromatic which will have a lot of values greater than 255. See GetStatistics/GetMinimum/GetMaximum functions.

Reply:
updating in new version